File: KafkaProducerExample/scalas/KafkaProducerExample/src/main/scala/com/dimanche/examples/KafkaProducerExample.scala
package com.dimanche.examples
import org.apache.log4j.{Level, Logger}
import java.util.Properties
import java.util.Date
import java.text.SimpleDateFormat

import org.apache.kafka.common.serialization.StringSerializer
import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord}

object KafkaProducerExample {

  def main(args: Array[String]): Unit = {

    Logger.getLogger("org").setLevel(Level.ERROR)

    val format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ssZ")

    val props = new Properties()
      props.put("bootstrap.servers", "localhost:9092")
      props.put("key.serializer", new StringSerializer().getClass.getName)
      props.put("value.serializer", new StringSerializer().getClass.getName)

    val producer = new KafkaProducer[String, String](props)

    val TOPIC ="topic_one"
    val KEY = TOPIC

    for(i <- 1 to 10) {

      val date = format.format(new Date())
      val value : String =s"""
          |{
          | "time": $date,
          | "data_type":  "Sample",
          | "value": "the current number is $i"
          |}
      """.stripMargin

      val producerRecord = new ProducerRecord(TOPIC, KEY, value)
      producer.send(producerRecord)

      Thread.sleep(1000)

    }

  }

}

File: 01_kafka_producer_example/java/KafkaProducerExample/src/main/java/com/dimanche/examples/KafkaProducerExample.java
package com.dimanche.examples;

import org.apache.kafka.clients.producer.KafkaProducer;
import org.apache.kafka.clients.producer.ProducerRecord;
import org.apache.kafka.common.serialization.StringSerializer;

import java.util.Properties;
import static java.lang.Thread.sleep;


public class KafkaProducerExample {

    static KafkaProducer<String, String> getProducer (){
        Properties props = new Properties();
        props.put("bootstrap.servers", "10.0.0.9:9092");
        props.put("acks", "all");
        props.put("retries", 0);
        props.put("batch.size", 16384);
        props.put("linger.ms", 1);
        props.put("buffer.memory", 33554432);
        props.put("key.serializer",   StringSerializer.class.getName());
        props.put("value.serializer", StringSerializer.class.getName());

        return new KafkaProducer<String, String>(props);
    }

    public static void main(String[] args) throws Exception{

        //Assign topicName to string variable
        String topicName = "topic_one";

        KafkaProducer<String, String> producer = getProducer();

        for(int id = 0; id < 10; id++) {
            String data = String.format("{ \"id\": %s, \"link\":\"http://trefle.io/api/subkingdoms/1\", \"name\":\"Tracheobionta\", \"slug\":\"tracheobionta\"}",  Integer.toString(id));

            ProducerRecord<String, String> producerRecord = new ProducerRecord<String, String>(topicName, Integer.toString(id), data);
            producer.send(producerRecord);
            sleep(3000);
        }

        System.out.println("Message sent successfully");
        producer.close();
    }
}